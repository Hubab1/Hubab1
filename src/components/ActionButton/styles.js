import {
    css
} from 'emotion';
import styled from '@emotion/styled';

export const root = css`
    border-radius: 21.5px !important;
    height: 45px;
`;

export const label = css`
    text-transform: none;
    font-size: 16px;
`;

export const mainDisabled = css`
    background-color: #D5D8DE !important;
    color: #ffffff !important;
    box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14), 0px 3px 1px -2px rgba(0,0,0,0.12) !important;
`;

export const ButtonContainer = styled.div(props => ({
    marginTop: props.marginTop || 0,
    marginBottom: props.marginBottom || 0,
    textDecoration: 'none',
    display: 'block',
}));
