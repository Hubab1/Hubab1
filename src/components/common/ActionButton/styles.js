import {
    css
} from 'emotion';
import styled from '@emotion/styled';

export const root = css`
    border-radius: 21.5px !important;
`

export const label = css`
    text-transform: none;
`

export const ButtonContainer = styled.div(props => ({
    marginTop: props.marginTop || 0,
    marginBottom: props.marginBottom || 0
}))
