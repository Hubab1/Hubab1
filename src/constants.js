export const community = {
    name: '555 Waverly'
};

export const Routes = {
    WELCOME: 'welcome',
    PROFILE: 'profile',
    LOGIN: 'login',
    SIGNUP: 'signup',
    TOS: 'terms',
    PROFILE_OPTIONS: 'profile/options',
    PETS: 'profile/pets'
}

export const ALL_ROUTES = [
    Routes.LOGIN, Routes.TOS, Routes.PROFILE_OPTIONS, Routes.PETS
]
