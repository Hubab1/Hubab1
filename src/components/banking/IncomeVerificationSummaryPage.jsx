import React, { useEffect, useState, useMemo, useCallback } from 'react';
import { connect } from 'react-redux';
import { generatePath } from 'react-router';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { css } from 'emotion';
import { Link } from 'react-router-dom';
import { Typography } from '@material-ui/core';
import InfoIcon from '@material-ui/icons/Info';
import ErrorIcon from '@material-ui/icons/Error';

import {
    ROUTES,
    ROLE_GUARANTOR,
    INCOME_TYPE_FINICITY_AUTOMATED,
    ALL_INCOME_OR_ASSET_TYPES,
    FINANCIAL_STREAM_STATUS_INCOMPLETE,
    FINANCIAL_STREAM_ASSET,
} from 'app/constants';
import { prettyCurrency } from 'utils/misc';
import SimplePopover from 'components/common/SimplePopover';
import ResetApplicantFinancials from './ResetApplicantFinancials';
import ActionButton from 'components/common/ActionButton/ActionButton';
import Capsule from 'components/common/Capsule/Capsule';
import ExistingItemsExpansionPanel from 'components/profile/options/ExistingItemsExpansionPanel';
import BankingContext from './BankingContext';
import BackLink from 'components/common/BackLink';
import GenericFormMessage from 'components/common/GenericFormMessage';
import { H1, H3 } from 'assets/styles';
import { styles, Spacer, infoIconRoot } from 'assets/styles';
import finance from 'assets/images/finance.png';
import piggyBank from 'assets/images/piggy-bank.png';

const SkinnyH1 = styled(H1)`
    width: 70%;
`;

const SpacedH3 = styled(H3)`
    margin-top: 15px;
    margin-bottom: 30px;
`;
const linkStyle = {
    textDecoration: 'underline',
    fontSize: 14,
};

const totals = css`
    text-align: left;
    border-top: 1px solid #eeeeee;
    padding-top: 15px;
`;

const totalsValue = css`
    font-size: 16px;
    font-weight: bold;
`;

const totalsP = css`
    margin-top: 0px;
    font-size: 14px;
`;

const incomeOrAssetItemWarning = css`
    display: flex;
    flex-flow: row nowrap;
    margin-bottom: 10px;

    svg {
        color: #fb6d68;
        margin-right: 10px;
    }

    span {
        font-size: 12px;
        color: #fb6d68;
    }
`;

export const getIncompleteFinancialSourceWarning = (source, isAsset) => {
    const { adjusted_amount, status } = source;
    const isIncomplete = status === FINANCIAL_STREAM_STATUS_INCOMPLETE;

    if (!isIncomplete) return null;

    let warning;
    if (adjusted_amount) {
        if (isAsset) {
            warning = `The documents for this asset source show a value of ${prettyCurrency(adjusted_amount)}.`;
        } else {
            warning = `The documents for this income source show earnings of ${prettyCurrency(adjusted_amount)}/year.`;
        }
    } else {
        warning = `This ${
            isAsset ? 'asset' : 'income'
        } source has been marked as having incorrect or insufficient documents.`;
    }

    return warning;
};

export function IncomeOrAssetItemWarning({ source, isAsset }) {
    const warning = getIncompleteFinancialSourceWarning(source, isAsset);
    if (!warning) return null;

    return (
        <div className={incomeOrAssetItemWarning}>
            <ErrorIcon />
            <span>{warning}</span>
        </div>
    );
}

IncomeOrAssetItemWarning.propTypes = {
    source: PropTypes.object.isRequired,
    isAsset: PropTypes.bool.isRequired,
};

export function IncomeOrAssetsItem({ source }) {
    const isAsset = source.stream_type === FINANCIAL_STREAM_ASSET;

    const getSourceLabel = useCallback((source) => {
        if (source.finicity_income_stream_id && source.other) {
            return source.other;
        }
        return ALL_INCOME_OR_ASSET_TYPES[source.income_or_asset_type]?.label;
    }, []);

    const getProofString = useCallback((source) => {
        if (source.income_or_asset_type === INCOME_TYPE_FINICITY_AUTOMATED) {
            return 'Linked bank account';
        }

        // find all document type labels
        const typesSet =
            source.uploaded_documents?.reduce((accum, doc) => {
                if (doc.type?.label) {
                    accum.add(doc.type.label);
                }
                return accum;
            }, new Set()) || new Set();

        // join labels together with comma
        const proofs = Array.from(typesSet).join(', ');
        if (!proofs) return 'None';
        return proofs;
    }, []);

    return (
        <div>
            <IncomeOrAssetItemWarning source={source} isAsset={isAsset} />
            <div>{getSourceLabel(source)}</div>
            <div className={styles.colorManatee}>{prettyCurrency(source.estimated_amount)}</div>
            <div className={styles.colorManatee}>
                {`Proof of ${isAsset ? 'asset' : 'income'}: ${getProofString(source)}`}
            </div>
            {source.income_or_asset_type !== INCOME_TYPE_FINICITY_AUTOMATED && (
                <>
                    <Spacer height={10} />
                    <Link
                        style={linkStyle}
                        to={generatePath(ROUTES.EDIT_MANUAL_FINANCIAL_SOURCE, {
                            id: source.id,
                        })}
                    >
                        Edit
                    </Link>
                    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                    <Link style={linkStyle} to={generatePath(ROUTES.REMOVE_FINANCIAL_SOURCE, { id: source.id })}>
                        Remove
                    </Link>
                </>
            )}
        </div>
    );
}

IncomeOrAssetsItem.propTypes = {
    source: PropTypes.object.isRequired,
};

export function IncomeVerificationSummaryPage(props) {
    const context = React.useContext(BankingContext);
    const [showResetFinancials, setShowResetFinancials] = useState(false);

    const hasIncompleteIncomeSources = useMemo(() => {
        if (!context.bankingData?.income_sources) {
            return false;
        }

        return context.bankingData.income_sources.some(({ status }) => status === FINANCIAL_STREAM_STATUS_INCOMPLETE);
    }, [context.bankingData]);

    const hasIncompleteAssetSources = useMemo(() => {
        if (!context.bankingData?.asset_sources) {
            return false;
        }

        return context.bankingData.asset_sources.some(({ status }) => status === FINANCIAL_STREAM_STATUS_INCOMPLETE);
    }, [context.bankingData]);

    const showIncompleteFinancialSourcesWarning = hasIncompleteIncomeSources || hasIncompleteAssetSources;

    const setScrollPosition = useCallback(() => {
        // taken from https://github.com/ReactTraining/react-router/issues/394#issuecomment-128148470
        window.location.hash = window.decodeURIComponent(window.location.hash);
        const scrollToAnchor = () => {
            const hashParts = window.location.hash.split('#');
            if (hashParts.length > 1) {
                const hash = hashParts[1];
                const hashElement = document.querySelector(`#${hash}`);
                if (!!hashElement) {
                    hashElement.scrollIntoView();
                }
            }
        };
        scrollToAnchor();
        window.onhashchange = scrollToAnchor;
    }, []);

    useEffect(() => {
        setScrollPosition();
    }, [setScrollPosition]);

    const hashValue = props.location?.hash?.substring?.(1) ?? '';

    const getIncomeRequirementText = useCallback((config, profile, applicant) => {
        if (!profile || !applicant || !config) return <div />;

        const { guarantor_income_requirement_multiplier, applicant_income_requirements } = config;

        const guarantor_income_amount = guarantor_income_requirement_multiplier * profile.unit.price;
        const applicant_income_amount = applicant_income_requirements * profile.unit.price;

        if (applicant.role === ROLE_GUARANTOR) {
            return (
                <p className={totalsP}>
                    {prettyCurrency(guarantor_income_amount)} is the required income for guarantors.
                    {
                        <SimplePopover
                            text={`The required income for a guarantor is ${guarantor_income_requirement_multiplier}x the monthly rent`}
                        >
                            <InfoIcon classes={{ root: infoIconRoot }} style={{ color: '#828796', width: 16 }} />
                        </SimplePopover>
                    }
                </p>
            );
        }

        return (
            <p className={totalsP}>
                {prettyCurrency(applicant_income_amount)} is the recommended household income.
                <SimplePopover
                    text={`Itâ€™s recommended that the yearly combined income of all applicants be ${applicant_income_requirements}x the monthly rent.`}
                >
                    <InfoIcon classes={{ root: infoIconRoot }} style={{ color: '#828796', width: 16 }} />
                </SimplePopover>
            </p>
        );
    }, []);

    const onContinue = useCallback(() => {
        context._nextRoute();
    }, [context]);

    const hasNotAddedFinancialSources =
        !context.bankingData?.asset_sources.length && !context.bankingData?.income_sources.length;
    const reportedNoIncomeAssets = context.bankingData?.reported_no_income_assets;
    const canContinue = !hasNotAddedFinancialSources || reportedNoIncomeAssets;

    if (showResetFinancials) {
        return (
            <ResetApplicantFinancials
                onSubmit={() => {
                    context.clearFinancialSources();
                    props.history.push(ROUTES.INCOME_AND_EMPLOYMENT);
                }}
                onCancel={() => setShowResetFinancials(false)}
            />
        );
    }

    return (
        <>
            <SkinnyH1>
                {reportedNoIncomeAssets ? `Confirm Income and Assets` : `Income and Asset Verification`}
            </SkinnyH1>
            <SpacedH3>
                {reportedNoIncomeAssets
                    ? `Easy, right? Now just review the info below.`
                    : `Add at least one income source or asset below.`}
            </SpacedH3>
            {showIncompleteFinancialSourcesWarning && (
                <GenericFormMessage
                    type="error"
                    messages={['Oops! We found some issues with one or more of your income/assets sources.']}
                />
            )}
            <Capsule
                name="income"
                prefix={<img alt="coin" src={finance} />}
                label="Income"
                buttonLabel="Add an Income Source"
                tip="TBD"
                route={ROUTES.MANUAL_INCOME_ENTRY_ADD_INCOME}
                expansionPanel={
                    <>
                        {context.bankingData?.income_total && (
                            <div className={totals}>
                                <div className={totalsValue}>
                                    {prettyCurrency(parseInt(context.bankingData?.income_total))} Total Annual Income
                                </div>
                                {getIncomeRequirementText(props.config, props.profile, props.applicant)}
                            </div>
                        )}
                        <ExistingItemsExpansionPanel
                            label="Income Source"
                            labelQuantity={context.bankingData?.income_sources.length}
                            defaultExpanded={hashValue === 'income' || hasIncompleteIncomeSources}
                        >
                            {context.bankingData?.income_sources?.map((source) => (
                                <IncomeOrAssetsItem key={source.id} source={source} />
                            ))}
                        </ExistingItemsExpansionPanel>
                    </>
                }
            />
            <Capsule
                name="asset"
                prefix={<img alt="piggy bank" src={piggyBank} />}
                label="Assets"
                buttonLabel="Add an Asset"
                tip="TBD"
                route={ROUTES.MANUAL_ASSET_ENTRY_ADD_ASSET}
                expansionPanel={
                    <>
                        {context.bankingData?.asset_total && (
                            <div className={totals}>
                                <div className={totalsValue}>
                                    {prettyCurrency(parseInt(context.bankingData?.asset_total))} Total Asset Balance
                                </div>
                                <p className={totalsP}>
                                    Weâ€™ll take this number into consideration in addition to or in place of income.
                                </p>
                            </div>
                        )}
                        <ExistingItemsExpansionPanel
                            label="Asset"
                            labelQuantity={context.bankingData?.asset_sources.length}
                            defaultExpanded={hashValue === 'asset' || hasIncompleteAssetSources}
                        >
                            {context.bankingData?.asset_sources?.map((source) => (
                                <IncomeOrAssetsItem key={source.id} source={source} />
                            ))}
                        </ExistingItemsExpansionPanel>
                    </>
                }
            />
            <ActionButton disabled={!canContinue} marginTop={68} marginBottom={20} onClick={onContinue}>
                Continue
            </ActionButton>
            {hasNotAddedFinancialSources ? (
                <BackLink to={ROUTES.INCOME_AND_EMPLOYMENT} />
            ) : (
                <Typography
                    classes={{ root: styles.cursor }}
                    onClick={() => setShowResetFinancials(true)}
                    color="primary"
                >
                    Start Income Verification Over
                </Typography>
            )}
        </>
    );
}

IncomeVerificationSummaryPage.propTypes = {
    config: PropTypes.object,
    profile: PropTypes.object,
    applicant: PropTypes.object,
    location: PropTypes.object,
    history: PropTypes.object,
};

const mapStateToProps = (state) => ({
    config: state.configuration,
    profile: state.renterProfile,
    applicant: state.applicant,
});

export default connect(mapStateToProps)(IncomeVerificationSummaryPage);
