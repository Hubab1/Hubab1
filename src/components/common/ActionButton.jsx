import React from 'react';

import Button from '@material-ui/core/Button';
import { root, label } from 'assets/emotion/ActionButton';

export default class ActionButton extends React.Component {
    render () {
        const { color, disabled, onClick, children } = this.props;
        return (
            <Button
                onClick={onClick}
                classes={{ root, label }}
                variant="contained"
                color={color || 'primary'}
                type="submit"
                disabled={disabled}
                fullWidth
            >
                {children}
            </Button>
        );
    }
}