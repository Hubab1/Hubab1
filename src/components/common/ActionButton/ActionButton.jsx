import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';

import { root, label, ButtonContainer } from './styles';

export default class ActionButton extends React.Component {
    render () {
        const { color, disabled, onClick, children, marginTop, marginBottom, variant } = this.props;
        return (
            <ButtonContainer marginTop={marginTop} marginBottom={marginBottom}>
                <Button
                    onClick={onClick}
                    classes={{ root, label }}
                    variant={variant}
                    color={color}
                    type="submit"
                    disabled={disabled}
                    fullWidth
                >
                    {children}
                </Button>
            </ButtonContainer>
        );
    }
}

ActionButton.propTypes = {
    color: PropTypes.string,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    marginTop: PropTypes.string,
    marginBottom: PropTypes.string,
    variant: PropTypes.string,
}

ActionButton.defaultProps = {
    variant: 'contained',
    color: 'primary',
}