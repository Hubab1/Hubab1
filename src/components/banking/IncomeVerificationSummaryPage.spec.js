import React from 'react';
import { shallow } from 'enzyme';
import { IncomeVerificationSummaryPage } from './IncomeVerificationSummaryPage';
import { ROLE_GUARANTOR, ROLE_PRIMARY_APPLICANT } from 'app/constants';
import ActionButton from 'components/common/ActionButton/ActionButton';

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useContext: () => ({
        bankingData: {
            income_sources: [
                {
                    id: 123,
                    income_or_asset_type: 105,
                    estimated_amount: 100000,
                    uploaded_documents: [{
                        id: 4,
                        filename: 'w2-1.pdf',
                        type: {
                            label: 'W2',
                            id: 1,
                        }
                    }]
                },
                {
                    id: 124,
                    income_or_asset_type: 135,
                    estimated_amount: 100000,
                    uploaded_documents: []
                },
                {
                    id: 125,
                    income_or_asset_type: 500,
                    estimated_amount: 100000,
                    uploaded_documents: [{
                        id: 1,
                        filename: 'w2-1.pdf',
                        type: {
                            label: 'W2',
                            id: 1,
                        }
                    }, {
                        id: 2,
                        filename: 'w2-2.pdf',
                        type: {
                            label: 'W2',
                            id: 1,
                        }
                    }, {
                        id: 3,
                        filename: 'w3.pdf',
                        type: {
                            label: 'W3',
                            id: 2,
                        }
                    }]
                }
            ],
            asset_sources: [
                {
                    id: 125,
                    income_or_asset_type: 510,
                    estimated_amount: 40000,
                    uploaded_documents: []
                }
            ],
            income_total: '100000.00',
            asset_total: '40000.00',
        },

    })
  }));

let defaultProps;
beforeEach(() => {
    defaultProps = {
        history: {
            push: jest.fn()
        },
        goBack: jest.fn(),
        applicant: {
            role: ROLE_PRIMARY_APPLICANT
        },
        config: {
            guarantor_income_requirement_multiplier: 80,
            applicant_income_requirements: 40
        },
        profile: {
            unit: {
                price: 1000
            }
        }

    };
});

it('matches snapshot with some financial sources data', () => {
    const wrapper = shallow(<IncomeVerificationSummaryPage {...defaultProps}/>);
    expect(
        wrapper.getElement()
    ).toMatchSnapshot();
});

it('matches snapshot with some financial sources data if guarantor', () => {
    defaultProps.applicant.role = ROLE_GUARANTOR;
    const wrapper = shallow(<IncomeVerificationSummaryPage {...defaultProps} />);
    expect(
        wrapper.getElement()
    ).toMatchSnapshot();
});
