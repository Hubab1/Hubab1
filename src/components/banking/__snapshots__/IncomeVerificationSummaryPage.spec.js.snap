// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with some financial sources data 1`] = `
<React.Fragment>
  <ForwardRef(render)>
    Income and Asset Verification
  </ForwardRef(render)>
  <ForwardRef(render)>
    Add at least one income source or asset below.
  </ForwardRef(render)>
  <Capsule
    buttonLabel="Add an Income Source"
    expansionPanel={
      <React.Fragment>
        <div
          className="css-1qtbm8m"
        >
          <div
            className="css-mt9voc"
          >
            $100,000
             Total Annual Income
          </div>
          <p
            className="css-1pqho16"
          >
            $40,000
             is the recommended household income.
            <SimplePopover
              text="It’s recommended that the yearly combined income of all applicants be 40x the monthly rent."
            >
              <UNDEFINED
                classes={
                  Object {
                    "root": "css-bmph1p",
                  }
                }
                style={
                  Object {
                    "color": "#828796",
                    "width": 16,
                  }
                }
              />
            </SimplePopover>
          </p>
        </div>
        <ExistingItemsExpansionPanel
          defaultExpanded={false}
          label="Income Source"
          labelQuantity={1}
        >
          <div>
            <div>
              Employment
            </div>
            <div
              className="css-dufvih"
            >
              $100,000
              /year
            </div>
            <ForwardRef(render)
              height={10}
            />
            <ForwardRef
              style={
                Object {
                  "fontSize": 14,
                  "textDecoration": "underline",
                }
              }
              to="/income-employment/manual/financial-sources/123/edit"
            >
              Edit
            </ForwardRef>
          </div>
        </ExistingItemsExpansionPanel>
      </React.Fragment>
    }
    label="Income"
    limitReached={false}
    name="income"
    prefix={
      <img
        alt="coin"
        src="finance.png"
      />
    }
    route="/income-employment/manual/add-income"
    tip="TBD"
  />
  <Capsule
    buttonLabel="Add an Asset"
    expansionPanel={
      <React.Fragment>
        <div
          className="css-1qtbm8m"
        >
          <div
            className="css-mt9voc"
          >
            $40,000
             Total Asset Balance
          </div>
          <p
            className="css-1pqho16"
          >
            We’ll take this number into consideration in addition to or in place of income.
          </p>
        </div>
        <ExistingItemsExpansionPanel
          defaultExpanded={false}
          label="Asset"
          labelQuantity={1}
        >
          <div>
            <div>
              Investment
            </div>
            <div
              className="css-dufvih"
            >
              $40,000
            </div>
            <ForwardRef(render)
              height={10}
            />
            <ForwardRef
              style={
                Object {
                  "fontSize": 14,
                  "textDecoration": "underline",
                }
              }
              to="/income-employment/manual/financial-sources/123/edit"
            >
              Edit
            </ForwardRef>
          </div>
        </ExistingItemsExpansionPanel>
      </React.Fragment>
    }
    label="Assets"
    limitReached={false}
    name="asset"
    prefix={
      <img
        alt="piggy bank"
        src="piggy-bank.png"
      />
    }
    route="/income-employment/manual/add-asset"
    tip="TBD"
  />
  <ActionButton
    color="primary"
    marginBottom={20}
    marginTop={68}
    onClick={[Function]}
    successGreen={false}
    variant="contained"
  >
    Continue
  </ActionButton>
  <BackLink
    to="/income-employment"
  />
</React.Fragment>
`;

exports[`matches snapshot with some financial sources data if guarantor 1`] = `
<React.Fragment>
  <ForwardRef(render)>
    Income and Asset Verification
  </ForwardRef(render)>
  <ForwardRef(render)>
    Add at least one income source or asset below.
  </ForwardRef(render)>
  <Capsule
    buttonLabel="Add an Income Source"
    expansionPanel={
      <React.Fragment>
        <div
          className="css-1qtbm8m"
        >
          <div
            className="css-mt9voc"
          >
            $100,000
             Total Annual Income
          </div>
          <p
            className="css-1pqho16"
          >
            $80,000
             is the required income for guarantors.
            <SimplePopover
              text="The required income for a guarantor is 80x the monthly rent"
            >
              <UNDEFINED
                classes={
                  Object {
                    "root": "css-bmph1p",
                  }
                }
                style={
                  Object {
                    "color": "#828796",
                    "width": 16,
                  }
                }
              />
            </SimplePopover>
          </p>
        </div>
        <ExistingItemsExpansionPanel
          defaultExpanded={false}
          label="Income Source"
          labelQuantity={1}
        >
          <div>
            <div>
              Employment
            </div>
            <div
              className="css-dufvih"
            >
              $100,000
              /year
            </div>
            <ForwardRef(render)
              height={10}
            />
            <ForwardRef
              style={
                Object {
                  "fontSize": 14,
                  "textDecoration": "underline",
                }
              }
              to="/income-employment/manual/financial-sources/123/edit"
            >
              Edit
            </ForwardRef>
          </div>
        </ExistingItemsExpansionPanel>
      </React.Fragment>
    }
    label="Income"
    limitReached={false}
    name="income"
    prefix={
      <img
        alt="coin"
        src="finance.png"
      />
    }
    route="/income-employment/manual/add-income"
    tip="TBD"
  />
  <Capsule
    buttonLabel="Add an Asset"
    expansionPanel={
      <React.Fragment>
        <div
          className="css-1qtbm8m"
        >
          <div
            className="css-mt9voc"
          >
            $40,000
             Total Asset Balance
          </div>
          <p
            className="css-1pqho16"
          >
            We’ll take this number into consideration in addition to or in place of income.
          </p>
        </div>
        <ExistingItemsExpansionPanel
          defaultExpanded={false}
          label="Asset"
          labelQuantity={1}
        >
          <div>
            <div>
              Investment
            </div>
            <div
              className="css-dufvih"
            >
              $40,000
            </div>
            <ForwardRef(render)
              height={10}
            />
            <ForwardRef
              style={
                Object {
                  "fontSize": 14,
                  "textDecoration": "underline",
                }
              }
              to="/income-employment/manual/financial-sources/123/edit"
            >
              Edit
            </ForwardRef>
          </div>
        </ExistingItemsExpansionPanel>
      </React.Fragment>
    }
    label="Assets"
    limitReached={false}
    name="asset"
    prefix={
      <img
        alt="piggy bank"
        src="piggy-bank.png"
      />
    }
    route="/income-employment/manual/add-asset"
    tip="TBD"
  />
  <ActionButton
    color="primary"
    marginBottom={20}
    marginTop={68}
    onClick={[Function]}
    successGreen={false}
    variant="contained"
  >
    Continue
  </ActionButton>
  <BackLink
    to="/income-employment"
  />
</React.Fragment>
`;
